docs/src/contributing_apub_api_outline.md

```diff
@@ -1,380 +1,592 @@
+    "@context": "https://www.w3.org/ns/activitystreams",
+    "id": "https://enterprise.lemmy.ml/private_message/34",
+    "type": "Note",
+    "attributedTo": "https://enterprise.lemmy.ml/u/picard",
+    "to": "https://voyager.lemmy.ml/u/janeway",
+    "content": "test",
+    "published": "2020-10-08T19:10:46.542820+00:00",
+    "updated": "2020-10-08T20:13:52.547156+00:00"
 }
 ```
 
-#### [Update](https://www.w3.org/TR/activitystreams-vocabulary/#dfn-update)
-```
+| Field Name | Mandatory | Description |
+|---|---|---|
+| `attributedTo` | ID of the user who created this private message |
+| `to` | ID of the recipient |
+| `content` | yes | The text of the private message |
+| `published` | no | Datetime when the message was created |
+| `updated` | no | Datetime when the message was edited (not present if it was never edited) |
+
+## Activities
+
+### Follow
+
+When the user clicks "Subscribe" in a community, a `Follow` is sent. The community automatically responds with an `Accept/Follow`.
+
+Sent by: User
+
+Sent to: Community
+
+```json
 {
-  "@context": "https://www.w3.org/ns/activitystreams",
-  "type": "Create",
-  "actor": id,
-  "object": comment_id, or post_id
-  TODO fields.
+    "@context": "https://www.w3.org/ns/activitystreams",
+    "id": "https://enterprise.lemmy.ml/activities/follow/2e4784b7-4edf-4fa1-a352-674d5d5f8891",
+    "type": "Follow",
+    "actor": "https://enterprise.lemmy.ml/u/picard",
+    "to": "https://ds9.lemmy.ml/c/main",
+    "object": "https://ds9.lemmy.ml/c/main"
 }
 ```
-#### [Read](https://www.w3.org/TR/activitystreams-vocabulary/#dfn-read)
-```
+
+| Field Name | Mandatory | Description |
+|---|---|---|
+| `actor` | yes | The user that is sending the follow request |
+| `object` | yes | The community to be followed |
+
+### Accept Follow
+
+Automatically sent by the community in response to a `Follow`. At the same time, the community adds this user to its followers list.
+
+Sent by: Community
+
+Sent to: User
+
+```json
 {
-  "@context": "https://www.w3.org/ns/activitystreams",
-  "type": "Read",
-  "actor": user_id
-  "object": post_id
+    "@context": "https://www.w3.org/ns/activitystreams",
+    "id": "https://ds9.lemmy.ml/activities/accept/5314bf7c-dab8-4b01-baf2-9be11a6a812e",
+    "type": "Accept",
+    "actor": "https://ds9.lemmy.ml/c/main",
+    "to": "https://enterprise.lemmy.ml/u/picard",
+    "object": {
+        "@context": "https://www.w3.org/ns/activitystreams",
+        "id": "https://enterprise.lemmy.ml/activities/follow/2e4784b7-4edf-4fa1-a352-674d5d5f8891",
+        "type": "Follow",
+        "object": "https://ds9.lemmy.ml/c/main",
+        "actor": "https://enterprise.lemmy.ml/u/picard"
+    }
 }
 ```
 
-### Communities
-#### [Create](https://www.w3.org/TR/activitystreams-vocabulary/#dfn-create)
-```
+| Field Name | Mandatory | Description |
+|---|---|---|
+| `actor` | yes | The same community as in the `Follow` activity |
+| `to` | no | Inbox of the user which sent the `Follow` |
```

Comment:
This is probably mandatory.

Reply:
No its [not used](https://github.com/LemmyNet/lemmy/blob/main/lemmy_apub/src/inbox/user_inbox.rs#L90). But it should say "ID of the user", not inbox.